From c9e0952fb9590893309d51d48617e998035c39c5 Mon Sep 17 00:00:00 2001
From: Jeffrey Ladouceur <Jeffrey.Ladouceur@freescale.com>
Date: Thu, 23 May 2013 14:14:10 -0400
Subject: [PATCH 190/987] fsl_qbman: initialise structures to zero

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

The cgr and cgrd are passed as parameters to the qman_query_cgr api.
Initialise to all zero before using them.

Signed-off-by: Jeffrey Ladouceur <Jeffrey.Ladouceur@freescale.com>
Change-Id: I3a6c3fdb0107ded82aec61842ae6463d4849512c
Reviewed-on: http://git.am.freescale.net:8181/2680
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Thorpe Geoff-R01361 <Geoff.Thorpe@freescale.com>
Reviewed-by: Wang Haiying-R54964 <Haiying.Wang@freescale.com>
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
---
 drivers/staging/fsl_qbman/qman_debugfs.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/staging/fsl_qbman/qman_debugfs.c b/drivers/staging/fsl_qbman/qman_debugfs.c
index b3b19ac..6b4dc046 100644
--- a/drivers/staging/fsl_qbman/qman_debugfs.c
+++ b/drivers/staging/fsl_qbman/qman_debugfs.c
@@ -485,7 +485,8 @@ static int query_cgr_show(struct seq_file *file, void *offset)
 	int i, j;
 	u32 mask;
 
-	memset(&cgr, 0, sizeof(struct qm_mcr_querycgr));
+	memset(&cgr, 0, sizeof(cgr));
+	memset(&cgrd, 0, sizeof(cgrd));
 	cgr.cgrid = query_cgr_data.cgid;
 	ret = qman_query_cgr(&cgr, &cgrd);
 	if (ret)
-- 
1.9.1

